clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -725,14 +725,21 @@ else if (value instanceof String)
                     if (value instanceof Class)
                         return value;
                     else if (value instanceof String) {
-                        ClassLoader contextOrKafkaClassLoader = Utils.getContextOrKafkaClassLoader();
-                        // Use loadClass here instead of Class.forName because the name we use here may be an alias
-                        // and not match the name of the class that gets loaded. If that happens, Class.forName can
-                        // throw an exception.
-                        Class<?> klass = contextOrKafkaClassLoader.loadClass(trimmed);
-                        // Invoke forName here with the true name of the requested class to cause class
-                        // initialization to take place.
-                        return Class.forName(klass.getName(), true, contextOrKafkaClassLoader);
+                        try {
+                            ClassLoader contextOrKafkaClassLoader = Utils.getContextOrKafkaClassLoader();
+                            // Use loadClass here instead of Class.forName because the name we use here may be an alias
+                            // and not match the name of the class that gets loaded. If that happens, Class.forName can
+                            // throw an exception.
+                            Class<?> klass = contextOrKafkaClassLoader.loadClass(trimmed);
+                            // Invoke forName here with the true name of the requested class to cause class
+                            // initialization to take place.
+                            return Class.forName(klass.getName(), true, contextOrKafkaClassLoader);
+                        } catch (ClassNotFoundException e) {
+                            // load class from kafkaClassLoader when it not in ContextClassLoader
+                            ClassLoader kafkaClassLoader = Utils.getKafkaClassLoader();
```

Comment:
How could you confirm the exception thrown is from `non-kafkaClassLoader`?

Reply:
It's use `ContextClassLoader` first in `Utils.getContextOrKafkaClassLoader`. And if `contextOrKafkaClassLoader` is `KafkaClassLoader`, it will be load class twice with failed. Whatever happens, an exception will thrown.